src/maigic_nyu/trello/api.py

```diff
@@ -0,0 +1,247 @@
+"""API module that exposes Trello functionality to users."""
+from datetime import datetime
+from typing import Any
+
+from src.maigic_nyu.trello._trello_manager import (  # Fix the import path
+    TrelloCard,
+    TrelloChecklist,
+    TrelloManager,
+    _TrelloAPIError,
+    _TrelloError,
+    _TrelloRateLimitError,
+)
+
+__all__ = [
+    "TrelloAPIError",
+    "TrelloCard",
+    "TrelloChecklist",
+    "TrelloError",
+    "TrelloRateLimitError",
+    "add_attachment",
+    "add_checklist_item",
+    "add_comment",
+    "add_label",
+    "archive_card",
+    "create_card",
+    "create_checklist",
+    "create_list",
+    "get_board_lists",
+    "get_list_name",
+    "move_card",
+    "search_cards",
+    "update_card_due_date",
+]
+
+# Re-export exceptions with public names
+TrelloError = _TrelloError
+TrelloAPIError = _TrelloAPIError
+TrelloRateLimitError = _TrelloRateLimitError
+
+# Global TrelloManager instance
+_trello = None
+
+def _get_trello() -> TrelloManager:
+    """Get or create the global TrelloManager instance."""
+    global _trello
+    if _trello is None:
+        _trello = TrelloManager()
+    return _trello
```

Comment:
this singleton pattern is better implemented in a class where the global variable is a `ClassVar`